main.py

```diff
@@ -1,60 +1,67 @@
+        keep_running = 0
 
-import analyze
-from analyze import *
- 
-def createParser():
+def main_parser_create():
     parser = argparse.ArgumentParser()
-    parser.add_argument('-iL', '--ifacelist', action='store_const', const=True, help='Get list of ifaces and exit') #Get list of ifaces and exit
-    parser.add_argument('iface', nargs='?', help='Custom iface for sniffing') #Custom iface for sniffing
+    parser.add_argument('-iL', '--ifacelist', action='store_const', const=True, help='Get list of ifaces and exit') # Get list of ifaces and exit
+    parser.add_argument('-i', '--iface', nargs='?', help='Custom iface for sniffing') #Custom iface for sniffing
     parser.add_argument('--nopcap', action='store_const', const=True, help='Don\'t create pcap file') #Don't create pcap file
-    parser.add_argument('-pcap', '--pcap', default='pcap.pcap', help='Custom pcap-file to write to') #Custom pcap-file to write to
+    parser.add_argument('-p', '--pcap', default='dump.pcap', help='Custom pcap-file to write to') #Custom pcap-file to write to
     parser.add_argument('-c', '--count', type=int, default=-1, help='How many packets will be captured. When non provided, keep capturing until program closed') #How many packets should we capture? Default: INF
-    
-    return parser    
-      
-def capt(sniffobj):
-    for plen, t, buf in sniffobj.capture():
-        try:
+
+    return parser
+
+def main_capture_packets(sniff_obj) -> None:
+    global keep_running
+
+    for plen, t, buf in sniff_obj.capture():
+        if (keep_running):  
             packet = getFriendlyPacket(plen, t, buf)
-            #print(t, "  ", packet['sip'], " ->  ", packet['dip'], " :   ", packet['data'])
-            print(t, packet)
-        except Exception:
-            print("\n")
-            print("Error occured while packet analyzing:")
-            print(sys.exc_info())  
-            print("\n")
-            continue
-
-def main_func(iface, of, c):
+        else:
+            sys.exit(0)
+
+def main_loop(iface, of, c) -> None:
     try:
         sniffobj = Sniff(iface, out_file=of, count=c)
-        capt(sniffobj)
+        main_capture_packets(sniffobj)
     except Exception:
-        print(sys.exc_info())   
-
+        print(sys.exc_info())
 
+def main_ifaces_banner() -> None:
+    print("Select interface from list below:")
+    print("+--- LIST ---+")
+    print("\n".join(get_iface_list()))
+    print("+------------+")
 
 if __name__ == '__main__':
-    parser = createParser()
+    signal.signal(signal.SIGINT, helper_signal_handler)
+    parser = main_parser_create()
     args = parser.parse_args()
+
     if args.ifacelist:
-        for i in get_iface_list():
-            print(i)
+        main_ifaces_banner()
         sys.exit()
+
     iface = args.iface or get_first_iface()
     pcap = args.pcap
-    if args.nopcap:
-        pcap = "pcap.pcap"
-    count=args.count
-    main_func(iface, pcap, count)
 
+    if args.nopcap:
+        pcap = "dump.pcap"
```

Comment:
Ну а чо, баги тоже надо отрефакторить. Не фиксить же.
```suggestion
        pcap = ""
```

Reply:
согласен, недосмотрел